.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,133 @@
+name: Build and Publish Docker Images
+
+# Trigger on pushes to master branch, new semantic version tags, and pull request updates
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - "main"
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+--.*"
```

Comment:
Had offline conversation, this versioning is `v[0-9]+.[0-9]+.[0-9]+--<crate-name>` to align with semver regex requirements.

Reply:
More explanation here: https://github.com/docker/metadata-action#typesemver

Reply:
Should this be a single hyphen or is the 2nd hyphen used to differentiate between a [pre-release tag](https://semver.org/#spec-item-9)?

Reply:
@joroshiba is the build and push action handling the scenario of only building a single image based on the git tag? e.g. `v0.0.6--astria-conductor` would only build the conductor image.

Is that something we need to handle here? or are we deciding on our semver strategy first before we implement it?

Reply:
Yeah used double hyphen to differentiate from the pre-release tags.

The tagging strategy is defined here, the filtering is done in the actual jobs. I might need to improve that a bit thinking about it now, but the idea is that if running because of a tag only run the relevant job, but we can add additional build jobs in here.